ynr/apps/elections/templates/elections/ballot_view.html

```diff
@@ -31,92 +31,323 @@
+      {% else %}
+        {% if candidates_locked %}
+          <p>
+            This list of candidates is now <strong>locked</strong>;
+            you can still update contact details of candidates, but
+            can't change the people standing in this constituency.
+          </p>
         {% endif %}
+      {% endif %}
 
+      {% if membership_edits_allowed %}
+        <p>
+          <a class="show-new-candidate-form button" href="{% url 'person-create' election=ballot.election.slug %}">
+            Add a new candidate
+          </a>
+        </p>
+        {% if sopn %}
+          <a href="{% url "bulk_add_from_sopn" ballot.election.slug ballot.post.id %}" class="button">
+            Add candidates from nomination paper in {{ballot.post.label}}
+          </a>
+        {% endif %}
+
+      {% elif not request.user.is_authenticated %}
+        <p>
+          <a href="{% url 'account_login' %}?next={{ request.path }}" class="show-new-candidate-form button">
+            Sign in to add a new candidate
+          </a>
+        </p>
+      {% endif %}
     </div>
-    <div class="columns large-3">
-        {% include "uk/data_timeline.html" %}
+
+    {% if membership_edits_allowed and add_candidate_form %}
+      <div class="candidates__new" {% if add_candidate_form.errors %}style="display: block"{% endif %}>
+        <h4>Add a new candidate</h4>
+
+        <form id="new-candidate-form" name="new-candidate-form" action="{% url 'person-create' election=ballot.election.slug %}" method="post">
+        {% with form=add_candidate_form identifiers_formset=identifiers_formset %}
+          {% include 'candidates/_person_form.html' %}
+        {% endwith %}
+        </form>
+      </div>
+    {% endif %}
+
+
+    {% if membership_edits_allowed %}
+
+
+    {% comment %}<h2>People not standing again</h2>
+
+    {{ people_not_standing }}
+{% endcomment %}
+    {% if candidates_might_stand_again %}
+
+
+    <div class="candidates__previous">
+      <h3>Is a candidate from an earlier election standing again?</h3>
+
+
+      <table class="table candidates-list">
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Party</th>
+            {% if user.is_authenticated %}
+            <th>Actions</th>
+            {% endif %}
+          </tr>
+        </thead>
+        <tbody>
+          {% for candidate in candidates_might_stand_again %}
+          <tr>
+            <td>
+              <a href="{{ candidate.person.get_absolute_url }}">
+                <img class="person-avatar" src="{{ candidate.person.get_display_image_url }}"/>
+                {% if candidate.result.is_winner %}<strong>{% endif %}
+                  {{ candidate.person.name }}
+                {% if candidate.result.is_winner %}</strong>{% endif %}
+              </a>
+            </td>
+            <td>{{ candidate.party.name }}</td>
+            <td>
+              <button class="button tiny js-toggle-source-confirmation standing">Standing again</button>
```

Comment:
A few lines up, we're only showing the "Actions" header if the user is authenticated, but then it looks like we're rendering the column/form content anyway, even if they're not.